sklearn/utils/tests/test_validation.py

```diff
@@ -356,13 +362,20 @@ def test_check_array():
                 assert X is X_checked
 
     # allowed sparse != None
-    X_csc = sp.csc_matrix(X_C)
-    X_coo = X_csc.tocoo()
-    X_dok = X_csc.todok()
-    X_int = X_csc.astype(int)
-    X_float = X_csc.astype(float)
 
-    Xs = [X_csc, X_coo, X_dok, X_int, X_float]
+    # try different type of sparse format
+    Xs = []
+    Xs.extend(
+        [
+            sparse_container(X_C)
+            for sparse_container in CSR_CONTAINERS
+            + CSC_CONTAINERS
+            + COO_CONTAINERS
+            + DOK_CONTAINERS
+        ]
+    )
+    Xs.extend([Xs[0].astype(int), Xs[0].astype(float)])
```

Comment:
I would rather never use the Python types as numpy dtypes. There are explicit and I am sure many readers would not necessarily understand which numpy dtype they correspond to on different platforms. Let's always use numpy dtypes explicitly:

```suggestion
    Xs.extend([Xs[0].astype(np.int64), Xs[0].astype(np.float64)])
```

but then I an not sure why calling `.astype(np.int64)` on something that is already of dtype int64 (since it derives from `np.arange`).